### PR TITLE
Add scripts to index jenkins job status

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit adds a script to index jenkins job details including status,
duration, build url, run id etc. into elasticsearch.

### CODE REVIEW

File: indexing/jenkins/index.sh [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  echo "Jenkins credentials are not defined, please check"
+  help
+  exit 1
+fi
+
+# Generate a uuid
+export UUID=$(uuidgen)
+
+# Elasticsearch and jenkins credentials
+export ES_SERVER=$ES_SERVER
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:40]: these exports are redundant

File: indexing/jenkins/index.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+export ES_INDEX=$ES_INDEX
+export JENKINS_USER=$JENKINS_USER
+export JENKINS_API_TOKEN=$JENKINS_API_TOKEN
+
+# Jenkins job info
+export JOB_NAME=$JOB_NAME
+export BUILD_NUMBER=$BUILD_NUMBER
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:47]: these exports are redundant

File: indexing/jenkins/index.sh [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+export BUILD_TAG=$JENKINS_BUILD_TAG
+export NODE_NAME=$JENKINS_NODE_NAME
+export BUILD_URL=$JENKINS_BUILD_URL
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:50]: can these be directly refereced?

File: indexing/jenkins/index.sh [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+infra=$(oc get nodes -l node-role.kubernetes.io/infra --no-headers=true | wc -l)
+cluster_version=$(oc get clusterversion -o jsonpath='{.items[0].status.desired.version}')
+network_type=$(oc get network cluster  -o jsonpath='{.status.networkType}')
+all=$(oc get nodes  --no-headers=true | wc -l)
+
+# Get the status and duration of the run
+JOB_STATUS=$(curl -k --user "$JENKINS_USER:$JENKINS_API_TOKEN" $BUILD_URL/api/json | jq '.result')
+JOB_DURATION=$(curl -k --user "$JENKINS_USER:$JENKINS_API_TOKEN" $BUILD_URL/api/json | jq '.duration')
+UPSTREAM_JOB=$(curl -k --user "$JENKINS_USER:$JENKINS_API_TOKEN" $BUILD_URL/api/json | jq '.actions[0].causes[].upstreamProject')
+UPSTREAM_JOB_BUILD=$(curl -k --user "$JENKINS_USER:$JENKINS_API_TOKEN" $BUILD_URL/api/json | jq '.actions[0].causes[].upstreamBuild')
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:70]: add the -r flag for raw output 
```
jq -r
```

File: indexing/jenkins/index.sh [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  # Index data into elasticsearch
+  curl -X POST -H "Content-Type: application/json" -H "Cache-Control: no-cache" -d '{
+    "uuid" : "'$UUID'",
+    "run_id" : "'${UPSTREAM_JOB}-${UPSTREAM_JOB_BUILD}'",
+    "platform": "'$platform'",
+    "master_count": '$masters',
+    "worker_count": '$workers',
+    "infra_count": '$infra',
+    "workload_count": '$workload',
+    "total_count": '$all',
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:85]: also in line 89

File: indexing/jenkins/index.sh [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    "network_type": "'$network_type'",
+    "cluster_version": "'$cluster_version'",
+    "total_count": '$all',
+    "build_number": '$BUILD_NUMBER',
+    "job_name": '$JOB_NAME',
+    "build_tag": "'$BUILD_TAG'",
+    "node_name": "'$NODE_NAME'",
+    "job_status": '$JOB_STATUS',
+    "build_url": "'$BUILD_URL'",
+    "upstream_job": '$UPSTREAM_JOB',
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:96]: "'$UPSTREAM_JOB'"

File: indexing/jenkins/index.sh [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    "upstream_job_build": '$UPSTREAM_JOB_BUILD',
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:97]:
```
"'$UPSTREAM_JOB_BUILD'"
```

File: indexing/jenkins/index.sh [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+    "worker_count": '$workers',
+    "infra_count": '$infra',
+    "workload_count": '$workload',
+    "total_count": '$all',
+    "cluster_name": "'$cluster_name'",
+    "network_type": "'$network_type'",
+    "cluster_version": "'$cluster_version'",
+    "total_count": '$all',
+    "build_number": '$BUILD_NUMBER',
+    "job_name": '$JOB_NAME',
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:91]:
```
"'$JOB_NAME'"
```

File: indexing/jenkins/index.sh [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    "build_tag": "'$BUILD_TAG'",
+    "node_name": "'$NODE_NAME'",
+    "job_status": '$JOB_STATUS',
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:94]:
```
"'$JOB_NAME'"
```

File: indexing/jenkins/index.sh [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    "cluster_name": "'$cluster_name'",
+    "build_tag": "'$BUILD_TAG'",
+    "node_name": "'$NODE_NAME'",
+    "job_status": '$JOB_STATUS',
+    "build_url": "'$BUILD_URL'",
+    "upstream_job": '$UPSTREAM_JOB',
+    "upstream_job_build": '$UPSTREAM_JOB_BUILD',
+    "job_duration": "'$JOB_DURATION'",
+    "timestamp": "'$timestamp'"
+    }' $ES_SERVER/$ES_INDEX/_doc/
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:128]: same substitutions as above

**Comment by mohit-sheth** [comment]
and example 
```
{
  "_index": "perf-scale-ci",
  "_type": "_doc",
  "_id": "Evt9CHcBS77rrkUzBbYU",
  "_version": 1,
  "_score": null,
  "_source": {
    "uuid": "751f0b40-e666-4a2p9-92be-b311fc683423",
    "run_id": "SCALE-CI-PIPELINE-GCP-4.7-17",
    "platform": "GCP",
    "master_count": 3,
    "worker_count": 3,
    "infra_count": 3,
    "workload_count": 1,
    "network_type": "OpenShiftSDN",
    "cluster_version": "4.7.29",
    "total_count": 10,
    "build_number": 155,
    "job_name": "INDEXER",
    "cluster_name": "perfciaws-mwfbd",
    "build_tag": "jenkins-INSTALL-26",
    "node_name": "thor",
    "job_status": "FAILURE",
    "build_url": "https://jenkins.scalelab.redhat.com/job/INSTALL/26/",
    "upstream_job": "SCALE-CI-PIPELINE-GCP-4.7",
    "upstream_job_build": 17,
    "job_duration": "5179",
    "timestamp": "2021-01-15T23:52:37.600"
  },
  "fields": {
    "timestamp": [
      "2021-01-15T23:52:37.600Z"
    ]
  },
  "sort": [
    1610754757600
  ]
}
```
from https://search-perfscale-pro-wxrjvmobqs7gsyi3xvxkqmn7am.us-west-2.es.amazonaws.com/_plugin/kibana/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-1y,to:now))&_a=(columns:!(_source),filters:!(),index:e695a170-5778-11eb-b9e3-49d5a71d05b5,interval:auto,query:(language:kuery,query:''),sort:!())

here we might want to remove 
```
    "build_number": 155,
    "job_name": "INDEXER",
```
as it doesn't provide any useful info
instead can we have 
```
    "build_number": 26,
    "job_name": "INSTALL",
```

**Comment by mohit-sheth** [comment]
also earlier couldn't find a way to scrape a build date from the clusterversion json
not every time the cluster version has a date like 
```
4.7.0-0.nightly-2021-01-19-095812
```
as seen in the above comment's example

**Comment by Harshith-umesh** [comment]
Results for e2e-benchmarking CI Tests
Workload                | Test                           | Result | Runtime  |
------------------------|--------------------------------|--------|----------|

File: indexing/jenkins/index.sh [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    "node_name": "'$JENKINS_NODE_NAME'",
+    "job_status": "'$JOB_STATUS'",
+    "build_url": "'$JENKINS_BUILD_URL'",
+    "upstream_job": "'$UPSTREAM_JOB'",
+    "upstream_job_build": "'$UPSTREAM_JOB_BUILD'",
+    "job_duration": "'$JOB_DURATION'",
+    "benchmark_name": "'$benchmark_name'",
+    "benchmark_status": "'$benchmark_status'",
+    "timestamp": "'$timestamp'"
+    }' $ES_SERVER/$ES_INDEX/_doc/
```
Comment (mohit-sheth) [inline indexing/jenkins/index.sh:95]: @mfleader please review this

File: indexing/jenkins/index.sh [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+set -exo pipefail
+
+# Help
+function help(){
+  printf "\n"
+  printf "Usage: export <options> $0\n"
+  printf "\n"
+  printf "Options supported, export them as environment variables:\n"
+  printf "\t ES_SERVER=str,                    str=elasticsearch server url, default: ""\n"
```
Comment (rsevilla87) [inline indexing/jenkins/index.sh:11]: @chaitanyaenr there're no default defined for several variables

File: indexing/jenkins/index.sh (hunk 4 above)
Comment (rsevilla87) [reply indexing/jenkins/index.sh]: We can also make only one request (curl) and reuse the output

File: indexing/jenkins/index.sh [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+else
+  curl -X POST -H "Content-Type: application/json" -H "Cache-Control: no-cache" -d '{
+    "uuid" : "'$UUID'",
+    "run_id" : "'${UPSTREAM_JOB}-${UPSTREAM_JOB_BUILD}'",
+    "platform": "'$platform'",
+    "master_count": "'$masters'",
+    "worker_count": "'$workers'",
+    "infra_count": "'$infra'",
+    "workload_count": "'$workload'",
+    "total_count": "'$all'",
```
Comment (rsevilla87) [inline indexing/jenkins/index.sh:106]: Dupe

**Review by mfleader — COMMENTED** [review]
Looks good to me. It does not seem to be missing any required fields.

**Review by mohit-sheth — APPROVED** [review]
LGTM

File: indexing/jenkins/index.sh (hunk 12 above)
Comment (chaitanyaenr) [reply indexing/jenkins/index.sh]: Fixed it, thanks.

File: indexing/jenkins/index.sh (hunk 13 above)
Comment (chaitanyaenr) [reply indexing/jenkins/index.sh:104]: Fixed it, thanks.

**Comment by mohit-sheth** [comment]
tested, works

**Pull request merged**